packages/manager/modules/sms/src/sms/batches/details/routing.js

```diff
@@ -0,0 +1,51 @@
+import moment from 'moment';
+
+import component from './telecom-sms-batches-details.component';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('sms.service.batches.details', {
+    url: '/:batchId',
+    views: {
+      'smsView@sms.service': {
+        component: component.name,
+      },
+    },
+    params: {
+      previousState: '^',
```

Comment:
Can't we use `$transition$.from()` or something like that instead ?

Reply:
Problem is that the details state has a child modal state (sms.service.batches.details.delete) which is called to open delete batch modal. If I use `$transition$.from()`, click on header back button may call the modal, instead of going back to parent page.

Reply:
**suggestion**: Can't the header back button always redirect to the same parent page ? which is `sms.service.batches`

Reply:
Why not, but here, the back button is set to go back to the page which has called the details page.
(It is not clear through this PR, but the details state can be called in two different points: through pending batches page or all batches page)